authentication/api_gateway/serializers.py

```diff
@@ -33,9 +34,14 @@ def create(self, validated_data):
         legal_address_data = validated_data.pop("legal_address")
         user_profile_data = validated_data.pop("user_profile", None)
         with transaction.atomic():
-            user.openedx_user.edx_username = username
             user.name = name
             user.save()
+            if user.openedx_user is None:
+                create_user(user, username)
+            else:
+                user.openedx_user.edx_username = username
+                user.openedx_user.save()
```

Comment:
I _think_ this can just be collapsed down to an unconditional call to `create_user` like this:
```suggestion
            create_user(user, username)
```